llama_index/llms/openai.py

```diff
@@ -103,6 +114,8 @@ def __init__(
             api_version=api_version,
         )
 
+        validate_openai_api_key(api_key=api_key)
```

Comment:
the reason we removed this was because subclasses of OpenAI might not require the OpenAI key to bet set. I think this will re-introduce that problem?

Reply:
ah yea that's true 😢

Reply:
I wonder if there's a way I can keep this, but avoid in subclasses. I find it really helpful

Reply:
I'll remove for the LLM, but keep for the embeddings (for now)

Reply:
removed for the LLM, kept for embeddings